### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by mimansha-swarup**
- login
- logout
- signup
- private routing
email: adarshbalika@gmail.com
pass: adarshBalika123

### CODE REVIEW

File: src/Context/authContext.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  const [authState, authDispatch] = useReducer(authReducer, initialState);
+
+  const { setToastData } = useToast();
+  const navigate = useNavigate();
+
+  const Login = async (email, password) => {
+    try {
+      const response = await axios.post(loginPostApi, { email, password });
+      console.log(response)
+      if (response.status === 200) {
```
Comment (suryxks) [inline src/Context/authContext.js:29]: remove console.log's before commit

**Review by suryxks — APPROVED** [review]
-code looks clean and readable 
-ui looks great 
-have a guest login or provide login credentials for reviewers so that we can test out the implemented features

File: src/Context/authContext.js (hunk 1 above)
Comment (mimansha-swarup) [reply src/Context/authContext.js:29]: thanks for reminding

**Review by sakshamAK — APPROVED** [review]
Great use of barral files, I love your clean code. Great work

**Pull request merged**